pkg/http/server.go

```diff
@@ -81,6 +90,43 @@ func (s *server) setupListener(opts Opts) error {
 	return nil
 }
 
+func setupUnixSocketListener(networkType string, address string) (net.Listener, error) {
+	socketOpts := strings.Split(address, ",")
+	if len(socketOpts) < 2 {
+		return net.Listen(networkType, address)
+	}
+
+	socketPath := socketOpts[0]
+	socketMode := os.FileMode(0o644)
```

Comment:
I'm not a big fan of setting a default value here, to be honest. Even if it's the current default of Go. However I'm not sure how else to do this nicely. Setting it as a reference + nil and checking later makes this look quite heavy

Reply:
Why not reference the value given to us by `umask`, the default mask?
Here's an example: https://go.dev/src/cmd/go/internal/toolchain/umask_unix.go

Reply:
Good point, I like that approach!

Reply:
Done